src/main/java/edu/kis/powp/jobs2d/plugin/FeatureManager.java

```diff
@@ -11,9 +11,127 @@ public static void registerFeature(FeaturePlugin plugin) {
         plugins.add(plugin);
     }
 
-    public static void initializeAll(Application app) {
+    public static void initializeAll(Application app) throws DependencyException {
+        System.out.println("Starting feature initialization with dependency checking...");
+
+        List<FeaturePlugin> orderedPlugins = resolveDependencies();
+
+        System.out.print("Initializing order: ");
+        int i = 0;
+        for (FeaturePlugin featurePlugin : orderedPlugins) {
+            i++;
+            System.out.print(featurePlugin.getClass().getSimpleName());
+            if (orderedPlugins.size() != i) System.out.print(" -> ");
+        }
+        System.out.println();
+
+        for (FeaturePlugin plugin : orderedPlugins) {
+            String featureName = plugin.getClass().getSimpleName();
+            System.out.println("Initializing feature: " + featureName);
+
+            try {
+                plugin.setup(app);
+                System.out.println("Successfully initialized: " + featureName);
+            } catch (Exception e) {
+                System.out.println("Failed to initialize " + featureName + ": " + e.getMessage());
+                throw new DependencyException("Feature initialization failed: " + featureName);
+            }
+        }
+
+        System.out.println("All features initialized successfully!");
+    }
+
+    private static List<FeaturePlugin> resolveDependencies() throws DependencyException {
+        Map<String, FeaturePlugin> featureMap = createFeatureMap();
+        Map<String, Set<String>> dependencies = buildFeaturesDependencies();
+
+        validateDependencies(featureMap, dependencies);
+
+        return sortFeatures(featureMap, dependencies);
+    }
+
+    private static Map<String, FeaturePlugin> createFeatureMap() {
+        Map<String, FeaturePlugin> featureMap = new HashMap<>();
         for (FeaturePlugin plugin : plugins) {
-            plugin.setup(app);
+            String featureName = plugin.getClass().getSimpleName();
+            featureMap.put(featureName, plugin);
+        }
+        return featureMap;
+    }
+
+    private static Map<String, Set<String>> buildFeaturesDependencies() {
+        Map<String, Set<String>> dependencies = new HashMap<>();
+
+        Set<String> workspaceDependencies = new HashSet<>();
+        workspaceDependencies.add("DriverFeature");
+        workspaceDependencies.add("DrawerFeature");
+        dependencies.put("WorkspaceFeature", workspaceDependencies);
+        dependencies.put("DriverFeature", new HashSet<>());
+        dependencies.put("DrawerFeature", new HashSet<>());
+        dependencies.put("CommandsFeature", new HashSet<>());
```

Comment:
Dependencies should be provided by features. FeatureManager should be ready to handle new features with new dependencies (not only hardcoded).